### ISSUE TITLE
Custom internal link style

### ISSUE THREAD

**Issue by torchatlas** [argenos/zotero-mdnotes#56]
**Is your feature request related to a problem? Please describe.**
Currently, for internal link style, we can choose between no link, [[wikilink]], and [markdown]\(link). However, I find a different link format to work better with my setup.

**Describe the solution you'd like**
Ability to customize the link format. The user would have a %c where the link's content would go.

**Additional context**
I already added this functionality on my local version of zotero-mdnotes. However, I haven't added a preference that enables/disables `lowerCaseDashTitle`, so this isn't perfect, since it's not ideal for wikilinks. Nor have I added a way to escape the %c with a backslash, in case anyone wants to. Still, it works!

**Example**
If someone wanted to add some emoji to their exported notes:

![image](https://user-images.githubusercontent.com/12666855/101258065-e7db7380-36ed-11eb-9715-5b6d617da0b2.png)

**Comment by argenos** [comment]
I'm not sure I would add this as a "Custom style" of links since what you want are wiki links with a custom prefix, right? I'd be willing to accept a PR that has a hidden preference for that. 

An alternative is to have something placeholder-style to format the links or with the current code (albeit more tedious) add the custom prefix as part of the placeholder itself.

**Comment by torchatlas** [comment]
> since what you want are wiki links with a custom prefix

The way I added the functionality, it supports anything. For example, I could have
`🔗 [[%c]] some random words [[%c]] ⭐ random emoji`

If the link's content was "Github", it would appear as:

`🔗 [[Github]] some random words [[Github]] ⭐ random emoji"`

The main change I made in adding this functionality is:

```diff
 function formatInternalLink(content, linkStyle) {
   linkStyle =
     typeof linkStyle !== "undefined" ? linkStyle : getPref("link_style");

   if (linkStyle === "wiki") {
     return `[[${content}]]`;
+   } else if (linkStyle === "custom") {
+     return getPref("custom_link_style").replace(/%c/g, content);
   } else if (linkStyle === "markdown") {
     return `[${content}](${lowerCaseDashTitle(content)})`;
   } else {
     return `${content}`;
   }
 }
```
with `custom_link_style` being what the user typed into the text box.

---

> An alternative is to have something placeholder-style

Not exactly sure what you mean by this. A placeholder in the templates? Because `%c` basically is a placeholder.

---

Of course, there could be a different way to implement this feature, like using the config editor.

**Comment by argenos** [comment]
Yeah, the way you are using %c is what I do with the placeholders in the templates. I would want to avoid adding new syntax to the plugin, so doing something similar to [field_contents](https://argentinaos.com/zotero-mdnotes/docs/formatting/#preference-value) would be better. 

I'm still unsure if this belongs in the link formatting preferences.

**Comment by torchatlas** [comment]
Well, this is interesting.

I managed to get this exact functionality by modifying a placeholder in the config editor!

I set `link_style` to `no-links`, and then just added what I wanted around `{{content}}`

For example, if I wanted a `▶ ` before my link, I could just do:

```
{"content":"{{bullet}} Topics: {{field_contents}}", "field_contents": "[[▶ {{content}}]]", "link_style": "no-links", "list_separator": ", "}
```

Of course, I have to make this change for multiple placeholders, but that's fine.
